resources/default.rb

```diff
@@ -0,0 +1,98 @@
+default_action :install
+
+provides :multipackage
+
+property :package_name, [String, Array], coerce: proc { |x| [x].flatten }, name_property: true
+property :version, [String, Array], coerce: proc { |x| [x].flatten }
+property :options, String
+property :timeout, [String, Integer]
+
+action :install do
+  do_action(:install)
+end
+
+action :remove do
+  do_action(:remove)
+end
+
+action :purge do
+  do_action(:purge)
+end
+
+action :reconfig do
+  do_action(:reconfig)
+end
+
+action :upgrade do
+  do_action(:upgrade)
+end
+
+action_class.send(:define_method, :my_declare_resource) do |type, name, &resource_attrs_block|
+  created_at = caller[0]
+
+  resource = Chef::ResourceBuilder.new(
+    type:                type,
+    name:                name,
+    created_at:          created_at,
+    params:              @params,
+    run_context:         run_context.parent_run_context,
+    cookbook_name:       cookbook_name,
+    recipe_name:         recipe_name,
+    enclosing_provider:  self.is_a?(Chef::Provider) ? self :  nil
+  ).build(&resource_attrs_block)
+  run_context.parent_run_context.resource_collection.insert(resource, resource_type: type, instance_name: name)
+  resource
+end
+
+action_class.send(:define_method, :do_action) do |action|
+  # @todo make sure package_names and versions have the same num of items
+  # (unless version is omitted)
+
+  #
+  # this whole code block could be modifiers to build_resource to build
+  # the resource in the parent run_context, with a flag to build it only
+  # if it does not already exist.
+  #
+
+  t = begin
+        run_context.parent_run_context.resource_collection.find(
+          multipackage_internal: "collected packages #{action}"
+        )
+      rescue Chef::Exceptions::ResourceNotFound
+        my_declare_resource(:multipackage_internal, "collected packages #{action}") do
+          package_name Array.new
+          version Array.new
+          action action
+        end
+      end
+
+  package_name.each_with_index do |pack, i|
+    if t.package_name.include?(pack)
+      # supress CHEF-3694 errors and be more useful about warning if there's only a reason
+      if options
+        Chef::Log.warn "ignoring options #{options} set on duplicated package #{pack}"
+      end
+      if timeout
+        Chef::Log.warn "ignoring timeout #{timeout} set on duplicated package #{pack}"
+      end
+      next
+    end
+    t.package_name.push(pack)
+    if version
+      t.version.push versions[i]
```

Comment:
Believe you are correct, that's referring to an old local variable that I removed....